### ISSUE TITLE
Add mechanism for providing details/resolution message to incompatibilities

### ISSUE THREAD

**Issue by Ortham** [loot/loot.github.io#55]
@LOOT

At the moment, incompatibility metadata generates messages of the form `This plugin is incompatible with X, but both are present.` 

Using the `display` key, a mod name or any other Markdown text can be used instead of the name of the plugin checked for, but while that's helpful for requirement messages (you can supply a link to the required mod that way), it's not so helpful for incompatibility messages.

I'm considering adding a new key to the File data structure that would allow another string to be appended to the message, so that additional details about resolving the incompatibility can be provided. For example, if there's a patch available, you could link to the patch in this additional message string.

```
name: foo.esp
display: Foo's Bar
condition: "not file(\"foo patch.esp\")"
detail: "A compatibility patch is available [here](http://www.example.com)."
```

This would then generate a message like `This plugin is incompatible with Foo's Bar, but both are present. A compatibility patch is available here.` (`here` would be a link). The `detail` string would be ignored for everything apart from incompatibility message generation.

(I think I'll also change the `but both are present` wording to something a little less clunky`.)

What do you think?

**Comment by Freso** [comment]
How would this function differently than `msg` entries in practice?

**Comment by WrinklyNinja** [comment]
Well, it would simplify the condition (since `msg` conditions would have to check for `foo.esp` and `foo patch.esp`), and make incompatibility handling more uniform.

**Comment by Freso** [comment]
The example in the original message still checks for `foo patch.esp` though, which might be why I'm not seeing the great benefit… :)

**Comment by WrinklyNinja** [comment]
Yeah, it is limited in that way...

**Comment by Kaenguru-Knobel-Kette** [comment]
LOOT currently lacks metadata for these kind of incompatibilities. E.g. if I decided to add Cutting Room Floor to my load order why would I know that I have to reinstall Relationship Dialogue Overhaul (RDO) to grab the patch as that's only mentioned on the mod page of RDO? (Well **I** know because I'm a patch maker but the average user certainly doesn't.) The same applies to many other mods like Immersive Citizens, Audio Overhaul for Skyrim, Immersive Sounds Compendium, Complete Crafting Overhaul Remade and so on.
I plan to add this metadata next and I'd be grateful if this enhancement issue is implemented.

To give a concret example of the benefits:
```
msg:
  - <<: *reqPatchForX
    condition: 'file("ESFCompanions.esp") and not active("Requiem_Minor_Arcana - ESF Companions.esp")'
    subs: [ 'Enhanced Skyrim Factions - The Companions Guild', 'http://www.nexusmods.com/skyrim/mods/72865' ]
```

```
inc:
  - name: 'ESFCompanions.esp'
    display: 'Enhanced Skyrim Factions - The Companions Guild'
    condition: 'not active("Requiem_Minor_Arcana - ESF Companions.esp")'
    detail: "A compatibility patch is available [here](http://www.nexusmods.com/skyrim/mods/72865)."
```
In pratice the outcome of a `msg:` and enhanced `inc:` entry are identical however the latter example is much neater both for adding and reading entries.

**Comment by Freso** [comment]
Okay, I'm slowly being won over by this. Would this work with multiple `inc: … - name: 'ESFCompanions.esp'` entries? See e.g., https://github.com/loot/fallout4/blob/9239c216fdcfca73d5fee09753a8da26ba046237/masterlist.yaml#L1907-L1921 where `DEF_WIDGETS_SURVIVAL1.esp` is incompatible with both `EveryonesBestFriend.esp` and `EveryonesBestFriend.esp` which are additionally incompatible with each other, so there's one patch for each of those, as well as a patch allowing all three to work together. In the proposed system, this would require 2 `inc` entries for each of these, 1 with a check that it's using both mods and 1 with a check that it's only using 1.

**Comment by WrinklyNinja** [comment]
> Okay, I'm slowly being won over by this. Would this work with multiple inc: … - name: > 'ESFCompanions.esp' entries? See e.g., https://github.com/loot/fallout4/blob/9239c216fdcfca73d5fee09753a8da26ba046237/masterlist.yaml#L1907-L1921 where DEF_WIDGETS_SURVIVAL1.esp is incompatible with both EveryonesBestFriend.esp and EveryonesBestFriend.esp which are additionally incompatible with each other, so there's one patch for each of those, as well as a patch allowing all three to work together. In the proposed system, this would require 2 inc entries for each of these, 1 with a check that it's using both mods and 1 with a check that it's only using 1.

Well, that's a pretty non-optimal case, I'd probably have an inc entry for each plugin linking to their patch, and then have a message that is displayed when both are present and their combination patch isn't, that links to the combination patch.

**Comment by sibir-ine** [comment]
I know this issue's a bit old, but I think it could be useful to be able to substitute a message alias into the `detail` key.

At the moment, if we want to convey that two plugins are incompatible & why, we add an `inc` entry & a message like `patchProvided`, `patchUnavailable`, `compatIssuesWithX`, etc. I think it'd be a bit clearer to the user that the details or solution provided are in reference to the incompatibility if the message was appended to it, & we'd get access to the existing translations for these aliases.

However, the current method is probably clear enough as is; I haven't seen anyone complain about this, but I could imagine it being a source of slight confusion for a handful of users. I don't know if the possibly non-existent benefit is really worth the effort required. 🤷

**Comment by Ortham** [comment]
@sibir-ine Could you give a couple of before / after examples to demonstrate the improvement?

**Comment by sibir-ine** [comment]
@Ortham Sure. I think the benefit is more obvious with an entry with multiple incompatibilities & messages, such as `Immersive Horses.esp` from the SSE masterlist. Right now, it looks like:

```yaml
    inc:
      - 'Convenient Horses.esp'
      - name: 'Relationship Dialogue Overhaul.esp'
        condition: 'active("Relationship Dialogue Overhaul.esp") and not active("Relationship Dialogue Overhaul - Immersive Horses Patch.esp") and not active("RDO - Immersive Horses Patch.esp")'
    msg:
      - *requiresMCM
      - <<: *patch3rdParty_LotD
        condition: 'active("LegacyoftheDragonborn.esm") and not active("DBM_ImmersiveHorses_Patch.esp")'
      - <<: *patch3rdParty
        subs:
          - 'Beyond Skyrim - Bruma SE'
          - '[Immersive Horses - Bruma Patch (BETA)](https://www.nexusmods.com/skyrimspecialedition/mods/21121/)'
        condition: 'active("BSHeartland.esm") and not active("Immersive Horses - BS Bruma Patch.esp")'
      - <<: *patch3rdParty
        subs:
          - 'Book Covers Skyrim'
          - '[Miscellaneous Patches](https://www.nexusmods.com/skyrimspecialedition/mods/17700/)'
        condition: 'active("Book Covers Skyrim.esp") and not active("Book Covers Skyrim - Immersive Horses Patch.esp")'
      - <<: *patch3rdParty
        subs:
          - 'Relationship Dialogue Overhaul'
          - '[Miscellaneous Patches](https://www.nexusmods.com/skyrimspecialedition/mods/17700/)'
        condition: 'active("Relationship Dialogue Overhaul.esp") and not active("Relationship Dialogue Overhaul - Immersive Horses Patch.esp") and not active("RDO - Immersive Horses Patch.esp")'
```

But with the `detail` key, it could look something like:

```yaml
    inc:
      - 'Convenient Horses.esp'
      - name: 'Relationship Dialogue Overhaul.esp'
        <<: *incPatch3rdParty
        subs:
          - 'Relationship Dialogue Overhaul'
          - '[Miscellaneous Patches](https://www.nexusmods.com/skyrimspecialedition/mods/17700/)'
        condition: 'active("Relationship Dialogue Overhaul.esp") and not active("Relationship Dialogue Overhaul - Immersive Horses Patch.esp") and not active("RDO - Immersive Horses Patch.esp")'
    msg:
      - *requiresMCM
      - <<: *patch3rdParty_LotD
        condition: 'active("LegacyoftheDragonborn.esm") and not active("DBM_ImmersiveHorses_Patch.esp")'
      - <<: *patch3rdParty
        subs:
          - 'Beyond Skyrim - Bruma SE'
          - '[Immersive Horses - Bruma Patch (BETA)](https://www.nexusmods.com/skyrimspecialedition/mods/21121/)'
        condition: 'active("BSHeartland.esm") and not active("Immersive Horses - BS Bruma Patch.esp")'
      - <<: *patch3rdParty
        subs:
          - 'Book Covers Skyrim'
          - '[Miscellaneous Patches](https://www.nexusmods.com/skyrimspecialedition/mods/17700/)'
        condition: 'active("Book Covers Skyrim.esp") and not active("Book Covers Skyrim - Immersive Horses Patch.esp")'
```

If multiple of these messages showed up with the current method, it would look like:

```
* You seem to be using Beyond Skyrim - Bruma SE, but you have not enabled a compatibility patch for this mod. A third party patch is available here: Immersive Horses - Bruma Patch (BETA)

* You seem to be using Book Covers Skyrim, but you have not enabled a compatibility patch for this mod. A third party patch is available here: Miscellaneous Patches

* You seem to be using Relationship Dialogue Overhaul, but you have not enabled a compatibility patch for this mod. A third party patch is available here: Miscellaneous Patches

* This plugin is incompatible with "Convenient Horses.esp", but both files are present.

* This plugin is incompatible with "Relationship Dialogue Overhaul.esp", but both files are present.
```

But with the `detail` key, it could look something like:

```
* You seem to be using Beyond Skyrim - Bruma SE, but you have not enabled a compatibility patch for this mod. A third party patch is available here: Immersive Horses - Bruma Patch (BETA)

* You seem to be using Book Covers Skyrim, but you have not enabled a compatibility patch for this mod. A third party patch is available here: Miscellaneous Patches

* This plugin is incompatible with "Convenient Horses.esp", but both files are present.

* This plugin is incompatible with "Relationship Dialogue Overhaul.esp", but both files are present. A third party patch is available here: Miscellaneous Patches
```

That way, some redundancy has been reduced in both the masterlist & the UI while making it a bit clearer what's being communicated to the user.

**Comment by Ortham** [comment]
Closing this since the functionality is included in LOOT's v0.17.0 release.
